### PR TITLE
FE Sandbox: Get plugin module path from bootdata instead of plugin settings

### LINKED ISSUE #84632: Frontend Sandbox: Public dashboard not loading
The bug:

Public dashboards with panels and datasources running inside the sandbox are not working because the endpoint to retrieve the module.js location from settings requires permissions. 

Expected behaviour:

The frontend sandbox should work as usual with public dashboards.

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Modifies where the frontend sandbox code loader gets the plugin information to load the plugin code.

Before this PR the sandbox would fetch the plugin settings from the grafana http api which could not only
generate additional http requests on load but possibly fail in specific auth scenarios such as publich 
dashboards.

In this PR the frontend sandbox now tries to get the modulejs path and type from the existing data
in grafanaBootData

**Why do we need this feature?**

So plugins can work in the frontend sandbox when loaded in public dashboards

**Who is this feature for?**

All plugins running inside the frontend sandbox

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/84632

### CODE REVIEW

File: public/app/features/plugins/sandbox/code_loader.ts [hunk 1]
```diff
@@ -88,3 +89,33 @@ export function patchSandboxEnvironmentPrototype(sandboxEnvironment: SandboxEnvi
     `${patchArrayVectorProrotypeMethods.toString()};${patchArrayVectorProrotypeMethods.name}()`
   );
 }
+
+export async function getPluginLoadData(pluginId: string): Promise<SandboxPluginMeta> {
```
Comment (jackw) [inline public/app/features/plugins/sandbox/code_loader.ts:93]: Should this function be async?

**Review by jackw — APPROVED** [review]
Left a comment about async usage other than that code LGTM. 🚀

File: public/app/features/plugins/sandbox/sandbox_plugin_loader.ts [hunk 2]
```diff
@@ -33,7 +31,7 @@ const pluginLogCache: Record<string, boolean> = {};
 export async function importPluginModuleInSandbox({ pluginId }: { pluginId: string }): Promise<System.Module> {
   patchWebAPIs();
   try {
-    const pluginMeta = await getPluginSettings(pluginId);
+    const pluginMeta = await getPluginLoadData(pluginId);
```
Comment (leventebalogh) [inline public/app/features/plugins/sandbox/sandbox_plugin_loader.ts:34]: Following [Jack's comment](https://github.com/grafana/grafana/pull/86702/files#r1576275937) we can probably remove the `await` here, too?

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

File: public/app/features/plugins/sandbox/sandbox_plugin_loader.ts (hunk 2 above)
Comment (academo) [reply public/app/features/plugins/sandbox/sandbox_plugin_loader.ts:34]: Indeed. I was trying a different approach and didn't remove async. Thanks for catching it.

**Pull request merged**